test/UnitTest.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using Xunit;
+
+namespace SecretHubTest
+{
+    public class TestSuite
+    {
+        [Fact]
+        public void TestReadSuccess()
+        {
+            SecretHub.SecretVersion secret = SecretHub.Client.Read("secrethub-xgo/dotnet/test-secret");
+            Assert.Equal("super_secret_value", secret.Data);
+        }
+
+        [Fact]
+        public void TestReadFail()
+        {
+            string errMessage = "cannot find secret: \"secrethub-xgo/dotnet/not-this-one\": Secret not found (server.secret_not_found) ";
+            var ex = Assert.Throws<ApplicationException>(() => SecretHub.Client.Read("secrethub-xgo/dotnet/not-this-one"));
+            Assert.Equal(ex.Message, errMessage);
+        }
+
+        [Fact]
+        public void TestReadSecretSuccess()
+        {
+            string secret = SecretHub.Client.ReadString("secrethub-xgo/dotnet/test-secret");
+            Assert.Equal("super_secret_value", secret);
+        }
+
+        [Fact]
+        public void TestReadSecretFail()
+        {
+            string errMessage = "cannot find secret: \"secrethub-xgo/dotnet/not-this-one\": Secret not found (server.secret_not_found) ";
+            var ex = Assert.Throws<ApplicationException>(() => SecretHub.Client.ReadString("secrethub-xgo/dotnet/not-this-one"));
+            Assert.Equal(ex.Message, errMessage);
+        }
+
+        [Fact]
+        public void TestResolveSuccess() {
+            Assert.Equal("super_secret_value", SecretHub.Client.Resolve("secrethub://secrethub-xgo/dotnet/test-secret"));  
+        }
+
+        [Theory]
+        [InlineData("secrethub-xgo/dotnet/test-secret", true)]
+        [InlineData("secrethub-xgo/dotnet/not-this-one", false)]
+        public void TestExists(string path, bool expectedTestResult) {
+            Assert.Equal(expectedTestResult, SecretHub.Client.Exists(path));
+        }

```

Comment:
Can you please add bad weather variant of these test-cases as well?

You could pass a string that is not a valid path to get an exception.

Reply:
Solved